Content.Server/_DV/CosmicCult/Abilities/CosmicConversionSystem.cs

```diff
@@ -0,0 +1,76 @@
+using Content.Server._DV.CosmicCult.Components;
+using Content.Server.Bible.Components;
+using Content.Server.Popups;
+using Content.Shared._DV.CosmicCult.Components;
+using Content.Shared.Damage;
+using Content.Shared.Mindshield.Components;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Popups;
+using Content.Shared.Stunnable;
+
+namespace Content.Server._DV.CosmicCult.Abilities;
+
+public sealed class CosmicConversionSystem : EntitySystem
+{
+    [Dependency] private readonly CosmicCultRuleSystem _cultRule = default!;
+    [Dependency] private readonly CosmicGlyphSystem _cosmicGlyph = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly SharedStunSystem _stun = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<CosmicGlyphConversionComponent, TryActivateGlyphEvent>(OnConversionGlyph);
+    }
+
+    private void OnConversionGlyph(Entity<CosmicGlyphConversionComponent> uid, ref TryActivateGlyphEvent args)
+    {
+        var possibleTargets = _cosmicGlyph.GetTargetsNearGlyph(uid, uid.Comp.ConversionRange, entity => HasComp<CosmicCultComponent>(entity));
+        if (possibleTargets.Count == 0)
+        {
+            _popup.PopupEntity(Loc.GetString("cult-glyph-conditions-not-met"), uid, args.User);
+            args.Cancel();
+            return;
+        }
+        if (possibleTargets.Count > 1)
+        {
+            _popup.PopupEntity(Loc.GetString("cult-glyph-too-many-targets"), uid, args.User);
+            args.Cancel();
+            return;
+        }
+
+        foreach (var target in possibleTargets) // FIVE GODDAMN if-statements? Yep. I know. Why? My brain doesn't have enough juice to write something more succinct.
```

Comment:
Code looks fine to me, I'd say the comment is unneeded

```suggestion
        foreach (var target in possibleTargets)
```